src/core/algorithms/nar/nar.cpp

```diff
@@ -0,0 +1,119 @@
+#include "nar.h"
+
+namespace model {
+std::string NAR::ToString() const {
+    std::string result;
+    result += std::to_string(qualities_.fitness);
+    result += " {";
+    size_t antecounter = 0;
+    for (auto const& [key, value] : ante_) {
+        if (antecounter > 0) {
+            result += ", ";
+        }
+        result += std::to_string(key);
+        result += ": ";
+        result += value->ToString();
+        antecounter++;
+    }
```

Comment:
Incorporate changes suggested by [Petua41](https://github.com/Petua41), but also use `std::ostringstream` for the `result`. It makes consecutive creation of strings more readable and efficient:
```cpp
std::string NAR::ToString() const {
    std::ostringstream result;
    result << qualities_.fitness << " {";
    for (auto it = ante_.begin(); it != ante_.end(); ++it) {
        if (it != ante_.begin()) {
            result << ", ";
        }
        result << it->first << ": " << it->second->ToString();
    }
    result << "} ===> {";
    for (auto it = cons_.begin(); it != cons_.end(); ++it) {
        if (it != cons_.begin()) {
            result << ", ";
        }
        result << it->first << ": " << it->second->ToString();
    }
    result << "} s: " << qualities_.support << " c: " << qualities_.confidence;
    return result.str();
}
```